### DISCUSSION TITLE
Create a `gtk-rs` base image for CI builds

### DISCUSSION THREAD

**Discussion by chriskilding** [Relm4/Relm4#551] [Ideas]
I enquired about the possibility of pinning the `gtk-rs` Docker base image to particular Fedora versions in https://github.com/gtk-rs/gtk-rs-core/pull/1172, hoping that we could also get some pinned versions of the `gtk-rs` image itself, which we could use to build Relm4 apps on CI.

Unfortunately the gtk-rs team has said that their Docker images are only intended for their internal CI jobs. Using the images for anything else is unsupported by them.

Therefore, if we want to support GitHub Actions builds of Relm4 apps with recent versions of GTK / GNOME (i.e. more recent than whatever the Ubuntu Actions base image provides), we'll need to make our own Docker base image and/or GitHub Action for building GTK4 apps.

Do you have any thoughts about how to take this forward?

**Comment by AaronErhardt** [comment]
The gtk-rs image is always very cutting edge (I think they install GTK from git), which is actually a quite good approach for CI in general. GTK is backwards compatible so the only problem you can have is accidentally enabling feature flags that require newer versions of GTK which are not available on your target system (a problem which also doesn't exist in flatpaks, which most Relm4 devs use). I'm not sure if we need our own image just for that...

**Reply by chriskilding** [reply]
The cutting edge aspect of it is not the problem; it's the lack of version pinning (which means your app's CI build is not repeatable/deterministic, as `latest` could change under the covers between one build and the next) and the fact that the GTK-RS developers have said they won't offer support for anyone using their images. This is why I think there needs to be an image that Relm4 app devs can depend on for their builds - to make a cohesive developer experience

**Reply by AaronErhardt** [reply]
It would definitely nice to have such a docker image, but it's not a high priority - for me at least. So I likely won't work on this myself, but I'd be happy to review and merge the necessary changes.
